raiden-derive/src/ops/batch_delete.rs

```diff
@@ -0,0 +1,170 @@
+                                    ].into_iter().collect(),
+                                };
+                                write_request.delete_request = Some(delete_request);
+                                write_request
+                            };
+
+                            write_requests.push(write_request);
+                        }
+
+                        write_requests
+                    };
+
+                    #builder_name {
+                        client: &self.client,
+                        write_requests,
+                        table_name: self.table_name(),
+                    }
+                }
+            }
+        }
+    } else {
+        quote! {
+            pub trait #trait_name {
+                fn batch_delete(&self, keys: std::vec::Vec<impl Into<#partition_key_type>>) -> #builder_name;
+            }
+
+            impl #trait_name for #client_name {
+                fn batch_delete(&self, keys: std::vec::Vec<impl Into<#partition_key_type>>) -> #builder_name {
+                    let write_requests = {
+                        let mut write_requests = vec![];
+                        for pk in keys.into_iter() {
+                            let pk_attr_value = pk.into().into_attr();
+
+                            let write_request = {
+                                let mut write_request = ::raiden::WriteRequest::default();
+                                let delete_request = ::raiden::DeleteRequest {
+                                    key: vec![
+                                        (stringify!(#partition_key_ident).to_string(), pk_attr_value),
+                                    ].into_iter().collect(),
+                                };
+                                write_request.delete_request = Some(delete_request);
+                                write_request
+                            };
+
+                            write_requests.push(write_request);
+                        }
+
+                        write_requests
+                    };
+
+                    #builder_name {
+                        client: &self.client,
+                        write_requests,
+                        table_name: self.table_name(),
+                    }
+                }
+            }
+        }
+    };
+
+    quote! {
+        #client_trait
+
+        pub struct #builder_name<'a> {
+            pub client: &'a ::raiden::DynamoDbClient,
+            pub write_requests: std::vec::Vec<::raiden::WriteRequest>,
+            pub table_name: String,
+        }
+
+        impl<'a> #builder_name<'a> {
+            pub async fn run(mut self) -> Result<::raiden::batch_delete::BatchDeleteOutput, ::raiden::RaidenError> {
+                // TODO: set the number of retry to 5 for now, which should be made more flexible
+                const RETRY: usize = 5;
+                const MAX_ITEMS_PER_REQUEST: usize = 25;
```

Comment:
For now, the limit of the number of items per request is taken into account, but the limit of whole data of a request (16MB) is not. Ideally this also should be taken into account but I think it's ok for the moment, because it seems to be very rare that the data size of 25 items exceeds 16MB in total.